components/storage/blockdevice/COMPONENT_SPIF/SPIFBlockDevice.h

```diff
@@ -16,6 +16,7 @@
 #ifndef MBED_SPIF_BLOCK_DEVICE_H
 #define MBED_SPIF_BLOCK_DEVICE_H
 
+#include "platform/SingletonPtr.h"
```

Comment:
Is that required?

Reply:
Yes, the static array of peripheral management blocks contains `Mutex`es, and if those weren't wrapped in a `SingletonPtr`, the array would be placed into RAM even if an image didn't use SPI.